massa-models/src/operation.rs

```diff
@@ -110,30 +159,59 @@ impl FromStr for OperationId {
                     .with_check(None)
                     .into_vec()
                     .map_err(|_| ModelsError::OperationIdParseError)?;
-                let u64_deserializer = U64VarIntDeserializer::new(Included(0), Included(u64::MAX));
-                let (rest, _version) = u64_deserializer
+                let operation_id_deserializer = OperationIdDeserializer::new();
+                let (rest, op_id) = operation_id_deserializer
                     .deserialize::<DeserializeError>(&decoded_bs58_check[..])
                     .map_err(|_| ModelsError::OperationIdParseError)?;
-                Ok(OperationId(Hash::from_bytes(
-                    rest.try_into()
-                        .map_err(|_| ModelsError::OperationIdParseError)?,
-                )))
+                if rest.is_empty() {
+                    Ok(op_id)
+                } else {
+                    Err(ModelsError::OperationIdParseError)
+                }
+            }
+            _ => Err(ModelsError::OperationIdParseError),
+        }
+    }
+}
+
+#[transition::impl_version(versions("0"))]
+impl FromStr for OperationId {
+    type Err = ModelsError;
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let mut chars = s.chars();
+        match chars.next() {
+            Some(prefix) if prefix == OPERATIONID_PREFIX => {
+                let data = chars.collect::<String>();
+                let decoded_bs58_check = bs58::decode(data)
+                    .with_check(None)
+                    .into_vec()
+                    .map_err(|_| ModelsError::OperationIdParseError)?;
+                let operation_id_deserializer = OperationIdDeserializer::new();
+                let (rest, op_id) = operation_id_deserializer
+                    .deserialize::<DeserializeError>(&decoded_bs58_check[..])
+                    .map_err(|_| ModelsError::OperationIdParseError)?;
+                if rest.is_empty() {
+                    Ok(op_id)
+                } else {
+                    Err(ModelsError::OperationIdParseError)
+                }
             }
             _ => Err(ModelsError::OperationIdParseError),
         }
     }
 }
```

Comment:
What do we even need that one for? The versioned FromStr doesn't bring anything extra to the table

Reply:
I'm forced otherwise I have this compile error : 
```
error[E0277]: the trait bound `OperationIdV0: FromStr` is not satisfied
  --> massa-models/src/operation.rs:47:74
   |
47 |     Clone, Copy, Eq, PartialEq, Ord, PartialOrd, Hash, SerializeDisplay, DeserializeFromStr,
   |                                                                          ^^^^^^^^^^^^^^^^^^ the trait `FromStr` is not implemented for `OperationIdV0`
   ```